### PR TITLE
dump tx output for reveal_pk transactions when requested

### PR DESCRIPTION
**Pull request by dan-u410**
## Describe your changes
It is currently not possible to generate the `reveal_pk` transaction offline. This PR will dump the transaction when requested. 

## Checklist before merging 
- [ ] If this PR has some consensus breaking changes, I added the corresponding `breaking::` labels
    - This will require 2 reviewers to approve the changes
- [ ] If this PR requires changes to the docs or specs, a corresponding PR is opened in the `namada-docs` repo
    - Relevant PR if applies: 
- [ ] If this PR affects services such as `namada-indexer` or `namada-masp-indexer`, a corresponding PR is opened in that repo
    - Relevant PR if applies:

### CODE REVIEW

**Comment by dan-u410** [comment]
there is some complexity to the implementation:
- to construct the `reveal_pk` transaction body, the code calls out to a helper `submit_reveal_aux`
- this method has some internal complexity such as checking network state and possibly returning `none` in cases where the transaction isnt needed
- as an advanced user, id prefer to always get an output when requested
- so I call out to `build_reveal_pk` directly to ensure we get a tx when the user requests to dump the transaction bytes

there are a few ways to implement this so wanted to add color to my choice. happy to change.

File: crates/apps_lib/src/client/tx.rs [hunk 1]
```diff
@@ -1153,12 +1153,31 @@ pub async fn submit_reveal_pk<N: Namada>(
 where
     <N::Client as namada_sdk::io::Client>::Error: std::fmt::Display,
 {
-    let tx_data =
-        submit_reveal_aux(namada, &args.tx, &(&args.public_key).into()).await?;
+    if args.tx.dump_tx || args.tx.dump_wrapper_tx {
+        if let Address::Implicit(ImplicitAddress(pkh)) =
+            &(&args.public_key).into()
+        {
+            let public_key = &namada
```
Comment (tzemanovic) [inline crates/apps_lib/src/client/tx.rs:1160]: we don't need to look-up the PK as we already have it in `args.public_key`

Comment (tzemanovic) [reply crates/apps_lib/src/client/tx.rs]: fixed in aaf76793f1521cdf3a34f166d166226df1229f64

**Review by tzemanovic — APPROVED** [review]
thx for the fix!

**Pull request merged**